pandas/core/dtypes/cast.py

```diff
@@ -110,7 +110,9 @@ def trans(x):  # noqa
                     np.prod(result.shape)):
                 return result
 
-        if issubclass(dtype.type, np.floating):
+        # don't convert bool to float GH16875
+        if issubclass(dtype.type, np.floating) and\
```

Comment:
We generally prefer to use parentheses instead of the backward-slash i.e.:
~~~python
if (issubclass(...) and
    ...):
~~~